test/foundry/Snapshot.t.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: CAL
+pragma solidity =0.8.17;
+
+import {CreateOffchainAssetReceiptVaultFactory} from "../../contracts/test/CreateOffchainAssetReceiptVaultFactory.sol";
+import {Test, Vm} from "forge-std/Test.sol";
+import {OffchainAssetReceiptVault} from "../../contracts/vault/offchainAsset/OffchainAssetReceiptVault.sol";
+import {OffchainAssetVaultCreator} from "./OffchainAssetVaultCreator.sol";
+
+contract SnapshotTest is Test, CreateOffchainAssetReceiptVaultFactory {
+    event Snapshot(uint256 id);
+    event SnapshotWithData(address sender, uint256 id, bytes data);
+
+    /// Test to checks Snapshot events are emitted
+    function testSnapshot(uint256 fuzzedKeyAlice, bytes memory data, string memory assetName, string memory assetSymbol)
+        external
+    {
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+        vault.grantRole(vault.ERC20SNAPSHOTTER(), alice);
```

Comment:
do we have a test that you can't do this without this role?

Reply:
yes, in Roles.t.sol rest